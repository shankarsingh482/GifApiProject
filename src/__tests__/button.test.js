
import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/buttonComponent';
import sinon from 'sinon';

//Testing Configuration.

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


//const mockFn =()=>('hey')

//Test Cases For button.
describe('Button',()=>
{
    it('Button is defined in search Header Component',() =>
    {
        expect(Button).toBeDefined();
    });
    it('Button  Should Render Correctly',()=>
    {
        const tree = shallow(
            <Button name='button test' />
        );
        expect(tree).toMatchSnapshot();
    });
    it('Button is clickable in Search Header Component',()=>
    {
        const onButtonClick = sinon.spy();

        const tree = shallow(
            <Button name='button test' onClick ={onButtonClick}/>
        )
        // const instance = tree.instance();
        // const goSpy = sinon.spy(instance,'moc');
        // instance.forceUpdate();
            //  sinon.assert.calledOnce(goSpy);
         tree.simulate('click');
         expect(onButtonClick).toBeCalled();
       
    })
});