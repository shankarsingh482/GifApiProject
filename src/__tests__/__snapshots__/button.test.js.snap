// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button  Should Render Correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "button test",
        " ",
      ],
      "className": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      "button test",
      " ",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "button test",
          " ",
        ],
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        "button test",
        " ",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Button Button is clickable in Search Header Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    handleClick={[Function]}
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "button test",
        " ",
      ],
      "className": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      "button test",
      " ",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "button test",
          " ",
        ],
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        "button test",
        " ",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
